File: Leetcode/Daily/28-05-2024/prob.cpp
class Solution {
public:
    int equalSubstring(string s, string t, int maxCost) {
        int n = s.size();
        vector<int>costs(n,0);
        for(int i=0;i<n;i++){
            costs[i] = abs(s[i]-t[i]);
        }
        int len = 0;
        int sum = 0;
        int l = 0;
        for(int r=0;r<n;r++){
            sum += costs[r];
            while(sum>maxCost and l<n){
                sum -= costs[l];
                l++;
            }
            len = max(len,r-l+1);
        }

        return len;

    }
};

File: Leetcode/Daily/28-05-2024/prob.go
func equalSubstring(s string, t string, maxCost int) int {
    n := len(s)
    sum := 0
    ans := 0
    l := 0
    for r:=0;r<n;r++{
        sum += abs(int(s[r])-int(t[r]))
        for sum > maxCost{
            sum -= abs(int(s[l])-int(t[l]))
            l+=1
        }

        ans = max(ans,r-l+1)
    }

    return ans
}

func abs(n int) int{
    if n>0 {return n}
    return -n
}